src/schema/v2/artist/helpers.ts

```diff
@@ -138,6 +147,19 @@ const splitEntities = (value, delimiter = "|") => {
   return entities
 }
 
+const artistNewFollowersPercentage = (artist) => {
+  if (
+    artist.follow_count === 0 ||
+    artist.follow_count - artist.last_year_follow_count === 0
+  )
+    return 0
+
+  return Math.floor(
+    (artist.last_year_follow_count * 100) /
+      (artist.follow_count - artist.last_year_follow_count)
+  )
+}
+
```

Comment:
**suggestion [blocking]:** export function and extract more local vars

I think this function should be exported so that it can be tested directly. It'll mean you can move some of the outer loop tests out of the tests for `getArtistInsights` and then can simplify the tests so that it's more clear what's happening here.

I took a stab at how you might want to extract some local vars here but I confess I did this all in the GitHub comment editor so it could be that I've made a typo or this isn't how Prettier would format - sorry about that!

```suggestion
export const computeFollowerGrowth = (artist) => {
  const thisYearCount = artist.follow_count
  const lastYearCount = artist.last_year_follow_count
  const followerGrowthAmount = thisYearCount - lastYearCount
  
  if (thisYearCount === 0 || followerGrowthAmount === 0)
    return 0

  const growthPercentage = Math.floor(
    (thisYearCount * 100) / followerGrowthAmount
  )
  
  return growthPercentage
}

```

So yeah what I'm doing here is trying to put some words around the concepts I see. I don't have a good sense for how much this code will change over time but I think it's worth starting out with a few more details in case we're back in here to update something.

*format via [Conventional Comments](https://conventionalcomments.org/)*

Reply:
Moved this function to the Gravity.